doc/source/development/rfc/rfc100_float16_support.rst

```diff
@@ -0,0 +1,132 @@
+.. _rfc-100:
+
+=============================
+RFC 100: Support float16 type
+=============================
+
+============== =============================================
+Author:        Erik Schnetter
+Contact:       eschnetter @ perimeterinstitute.ca
+Started:       2024-Jun-05
+Status:        Proposed
+Target:        GDAL 3.10
+============== =============================================
+
+Summary
+-------
+
+This RFC adds support for the IEEE 16-bit floating point data type
+(aka ``half``, ``float16``). It adds a new pixel data type
+``GDT_Float16``. Where supported by the compiler, it also adds a new
+C/C++ type alias ``GFloat16``. Even when the local system does not
+support ``GFloat16``, datasets of this type can still be accessed by
+converting from/to ``float``.
+
+Some drivers, in particular Zarr, will be extended to support this
+type. Drivers that do not support this type will be checked to ensure
+they report an appropriate errors when necessary.
+
+Motivation
+----------
+
+- The float16 type is increasingly common. It is supported by GPUs, by
+  modern CPUs, and by modern compilers (e.g. GCC 12).
+
+- Using ``uint16`` values to shuffle ``float16`` values around is
+  inconvenient and awkward. There is no mechanism in GDAL to mark
+  attributes or datasets which ``uint16`` values should be interpreted
+  as ``float16``.
```

Comment:
It should be mentioned here that at least the HDF5, GTiff and Zarr drivers already handle float16, but by exposing it as float32, using software conversion routines in port/cpl_float.h.

For HDF5, older version of he HDF5 library, do that conversion internally. Most recent ones, when built with float16 support, directly return the float16 values, and GDAL does the the float16<-->float32 conversion (https://github.com/OSGeo/gdal/pull/10052)

Reply:
In the case where the GDAL build doesn't support native float16, we'd still want the current behaviour of converting float16 <--> float32 for backward compatibility

Reply:
Added.